pkg/apis/storage/validation/validation.go

```diff
@@ -397,14 +397,16 @@ func ValidateCSIDriver(csiDriver *storage.CSIDriver) field.ErrorList {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, apivalidation.ValidateCSIDriverName(csiDriver.Name, field.NewPath("name"))...)
 
-	allErrs = append(allErrs, validateCSIDriverSpec(&csiDriver.Spec, field.NewPath("spec"))...)
+	allErrs = append(allErrs, validateCSIDriverSpecWithFSGroup(&csiDriver.Spec, field.NewPath("spec"))...)
 	return allErrs
 }
 
 // ValidateCSIDriverUpdate validates a CSIDriver.
 func ValidateCSIDriverUpdate(new, old *storage.CSIDriver) field.ErrorList {
-	allErrs := ValidateCSIDriver(new)
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateCSIDriverName(new.Name, field.NewPath("name"))...)
 
+	allErrs = append(allErrs, validateCSIDriverSpec(&new.Spec, field.NewPath("spec"))...)
```

Comment:
hmm, while this indeed does the job - I was thinking something like:

```
// validate name
allErrs = append(allErrs, apivalidation.ValidateCSIDriverName(new.Name, field.NewPath("name"))...)
# and then just ensure that CSIDriver's spec is immutable
	// Spec is read-only
	// If this ever relaxes in the future, make sure to increment the Generation number in PrepareForUpdate
	if !apiequality.Semantic.DeepEqual(old.Spec, new.Spec) {
		allErrs = append(allErrs, field.Invalid(field.NewPath("spec"), new.Spec, "field is immutable"))
	}
	return allErrs
```

This way we don't validate spec at all beyond comparing new and old spec and ensuring spec can't be modified.

Reply:
I *think* I added this as https://github.com/kubernetes/kubernetes/pull/95739/commits/a8394a43a6ac3861cf86ab849d74fc15c06666bd .

Note that if we modify the spec, such as to introduce the FSGroupPolicy, then Update will fail due to the spec being immutable. I'm not certain if this is what you were requesting, so if this is incorrect feel free to let me know.